core/src/services/config.rs

```diff
@@ -0,0 +1,2047 @@
+        let mut ds = f.debug_struct("Config");
+
+        ds.field("root", &self.root);
+        ds.field("email", &self.email);
+
+        ds.finish()
+    }
+}
+
+/// Config for Libsql services support.
+#[derive(Default, Serialize, Deserialize, Clone, PartialEq, Eq)]
+#[serde(default)]
+#[non_exhaustive]
+pub struct LibsqlConfig {
+    /// Connection string for libsql service.
+    pub connection_string: Option<String>,
+    /// Authentication token for libsql service.
+    pub auth_token: Option<String>,
+
+    /// Table name for libsql service.
+    pub table: Option<String>,
+    /// Key field name for libsql service.
+    pub key_field: Option<String>,
+    /// Value field name for libsql service.
+    pub value_field: Option<String>,
+    /// Root for libsql service.
+    pub root: Option<String>,
+}
+
+impl Debug for LibsqlConfig {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut ds = f.debug_struct("LibsqlConfig");
+        ds.field("connection_string", &self.connection_string)
+            .field("table", &self.table)
+            .field("key_field", &self.key_field)
+            .field("value_field", &self.value_field)
+            .field("root", &self.root);
+
+        if self.auth_token.is_some() {
+            ds.field("auth_token", &"<redacted>");
+        }
+
+        ds.finish()
+    }
+}
+
+/// Config for MemCached services support
+#[derive(Default, Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+#[serde(default)]
+#[non_exhaustive]
+pub struct MemcachedConfig {
+    /// network address of the memcached service.
+    ///
+    /// For example: "tcp://localhost:11211"
+    pub endpoint: Option<String>,
+    /// the working directory of the service. Can be "/path/to/dir"
+    ///
+    /// default is "/"
+    pub root: Option<String>,
+    /// Memcached username, optional.
+    pub username: Option<String>,
+    /// Memcached password, optional.
+    pub password: Option<String>,
+    /// The default ttl for put operations.
+    pub default_ttl: Option<Duration>,
+}
+
+/// Config for memory.
+#[derive(Default, Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+#[serde(default)]
+#[non_exhaustive]
+pub struct MemoryConfig {
+    /// root of the backend.
+    pub root: Option<String>,
+}
+
+/// Config for mini-moka support.
+#[derive(Default, Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+#[serde(default)]
+#[non_exhaustive]
+pub struct MiniMokaConfig {
+    /// Sets the max capacity of the cache.
+    ///
+    /// Refer to [`mini-moka::sync::CacheBuilder::max_capacity`](https://docs.rs/mini-moka/latest/mini_moka/sync/struct.CacheBuilder.html#method.max_capacity)
+    pub max_capacity: Option<u64>,
+    /// Sets the time to live of the cache.
+    ///
+    /// Refer to [`mini-moka::sync::CacheBuilder::time_to_live`](https://docs.rs/mini-moka/latest/mini_moka/sync/struct.CacheBuilder.html#method.time_to_live)
+    pub time_to_live: Option<Duration>,
```

Comment:
`Duration`'s default serde is weird (things like `{ "secs": 1, "nanos": 0 }`.

It's better to use [`humantime-serde`](https://docs.rs/humantime-serde/latest/humantime_serde/) but that should be a follow up and breaking change.

Reply:
I'm considering using an `int` directly instead of exposing `Duration`. What do you think?

Reply:
Somehow reasonable. In milliseconds precison?